### ISSUE TITLE
AppLovinMediationYandexAdapter causes damaged builds

### ISSUE THREAD

**Issue by PaXLiCh** [AppLovin/AppLovin-MAX-Unity-Plugin#225]
### MAX Plugin Version

latest

### Unity Version

2020.3.40f1

### Device/Platform Info

iOS target 16, Xcode 14.2

### Current Behavior

Upload fails with errors
`The Info.plist in the package must contain the CFBundleVersion key. Unable to validate your application.`
`The Info.plist in the package must contain the CFBundleShortVersionString. Unable to validate your application.`
If open xcworkspace in Xcode Main target Unity-iPhone does not have version strings, version code, display name, category.
during "archiving" we can get errors
`warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'UnityFramework' from project 'Unity-iPhone')`

`warning: Run script build phase 'Crashlytics Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'UnityFramework' from project 'Unity-iPhone')`

We assume that the problem with YandexMobileAds dependency library. This library sets the build settings named GENERATE_INFOPLIST_FILE to YES in the cocoapod settings, this causes the Info.plist already created in the project to be override with the existing Xcode build settings.

Related Issue: https://github.com/yandexmobile/yandex-ads-sdk-ios/issues/142

### Expected Behavior

Successful build, archiving and validating.

### How to Reproduce

Just add mediation and export Xcode project, look at main target Unity-iPhone.

**Comment by santoshbagadi** [comment]
Hi @PaXLiCh , since the `GENERATE_INFOPLIST_FILE` is coming from their podspec itself, we won't be able to do much. As there is already an open issue against their repo, we'll certify their latest SDK once it is fixed. In the meantime, we'll see if we can downgrade to a version that doesn't have this setting.

**Comment by dbystr** [comment]
Hi [@santoshbagadi](https://github.com/santoshbagadi) [PaXLiCh](https://github.com/PaXLiCh), please advice what is the fastest way to repeat the issue with the latest version of Unity?

**Comment by mobile-ads-github** [comment]
Here is the quick fix which addresses this and related issues when trying to generate Demo iOS project.

**1. Info.plist**

In the file [DemoApp/Assets/MaxSdk/Scripts/Editor/MaxPostProcessBuildiOS.cs](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/blob/09cbac40b66c21ce80c72e5c5c2cb1961f3ca432/DemoApp/Assets/MaxSdk/Scripts/Editor/MaxPostProcessBuildiOS.cs#L121) right before
```
project.WriteToFile(projectPath);
```
insert the line
```
project.SetBuildProperty(unityMainTargetGuid, "GENERATE_INFOPLIST_FILE", "NO");
```


**2. Version mismatch**

In the file [DemoApp/ProjectSettings/ProjectSettings.asset](https://st.yandex-team.ru/ADLIB-14135#:~:text=https%3A//github.com/AppLovin/AppLovin%2DMAX%2DUnity%2DPlugin/blob/09cbac40b66c21ce80c72e5c5c2cb1961f3ca432/DemoApp/ProjectSettings/ProjectSettings.asset%23L181) change
```
iOSTargetOSVersionString: 11.0
```
to
```
iOSTargetOSVersionString: 12.0
```

After these two steps your project should be able to build to Xcode:

**Unity**: File > Build Settings… > iOS > Build


**3. Disable bitcode** (if needed)

**Xcode:**
- Unity-iPhone > _your_device_
- PROJECT > Unity-iPhone > Build Settings > Enable Bitcode: No
- TARGETS > Unity-iPhone > Build Settings > Enable Bitcode: No


**4. Automatically manage signing** (if needed)
- TARGETS > Unity-iPhone > Signing & Capabilities > Automatically manage signing: checked


**5. Change team** (if needed)
- TARGETS > Unity-iPhone > Signing & Capabilities > Team > _your_team_

After these additional steps you should be able to compile demo project with Product > Build.

[@santoshbagadi] Please advice if you would like to work on update addressing these changes. If not, we could start working on update for item 1 (but won't be able to address 2, 3, and 4).

**Comment by santoshbagadi** [comment]
@mobile-ads-github thank you for sharing this fix. We'll try and get this into our next release.

**Comment by santoshbagadi** [comment]
Hi all, we’ve released a fix for this in our latest plugin [version 5.8.3](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/releases/tag/release_5_8_3).

**Comment by pingvigames** [comment]
Have you tested version 5.8.3 with Yandex adapter? Upgrading to the new version did not help us. We still have "The Info.plist in the package must contain the CFBundleVersion key. Unable to validate your application."
Try to build with sdk 5.8.3 + yandex adapter + fb sdk

**Comment by santoshbagadi** [comment]
@pingvigames Thank you for bringing this to our attention and apologies for the inconvenience. We'll release a fix for this in our next release.

**Comment by santoshbagadi** [comment]
@pingvigames we've released a fix for this in our plugin version [5.9.0](https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/releases/tag/release_5_9_0)
